demos/coreMQTT/mqtt_demo_mutual_auth.c

```diff
@@ -642,36 +649,65 @@ int RunCoreMqttMutualAuthDemo( bool awsIotMqttMode,
 }
 /*-----------------------------------------------------------*/
 
-static int32_t prvGenerateRandomNumber()
+static BaseType_t prvBackoffForRetry( BackoffAlgorithmContext_t * pxRetryParams )
 {
-    uint32_t ulRandomNum;
+    BaseType_t xReturnStatus = pdFAIL;
+    uint16_t usNextRetryBackOff = 0U;
+    BackoffAlgorithmStatus_t xBackoffAlgStatus = BackoffAlgorithmSuccess;
+
+    /**
+     * To calculate the backoff period for the next retry attempt, we will
+     * generate a random number to provide to the backoffAlgorithm library.
+     *
+     * Note: The PKCS11 module is used to generate the random number as it allows access
+     * to a True Random Number Generator (TRNG) if the vendor platform supports it.
+     * It is recommended to use a random number generator seeded with a device-specific
+     * entropy source so that probability of collisions from devices in connection retries
+     * is mitigated.
+     */
+    uint32_t ulRandomNum = 0;
 
-    /* Use the PKCS11 module to generate a random number. */
     if( xPkcs11GenerateRandomNumber( ( uint8_t * ) &ulRandomNum,
                                      ( sizeof( ulRandomNum ) == pdPASS ) ) )
```

Comment:
and here?  Now I'm beginning to wonder if we couldn't refactor this to be more DRY?

Reply:
The `xPkcs11GenerateRandomNumber` function is obtained from `demos/pkcs11_helpers/pkcs11_helpers.c` file. 
Commonizing the `prvBackoffForRetry` across the MQTT demos will increase the scope of this PR as adding a new `demos/coreMQTT/mqtt_common_utils.c` file will entail updating all project files. 
I prefer not increasing the scope of this PR..